.github/workflows/release-go.yml

```diff
@@ -28,12 +27,59 @@ jobs:
           cp ./piper_master-darwin.arm64 ./piper-darwin.arm64
           npm install semver --quiet
           echo "PIPER_version=v$(node_modules/.bin/semver -i minor $(curl --silent "https://api.github.com/repos/$GITHUB_REPOSITORY/releases/latest" | jq -r .tag_name))" >> $GITHUB_ENV
+
       - uses: SAP/project-piper-action@master
-        name: 'publish Linux master binary'
+        name: Publish prerelease
         with:
           piper-version: master
           command: githubPublishRelease
-          flags: --token ${{ secrets.GITHUB_TOKEN }} --assetPathList ./piper_master --assetPathList ./piper --assetPathList ./piper_master-darwin.x86_64 --assetPathList ./piper-darwin.x86_64 --assetPathList ./piper_master-darwin.arm64 --assetPathList ./piper-darwin.arm64
+          flags: >
+            --preRelease true
+            --token ${{ secrets.GITHUB_TOKEN }}
+            --assetPathList ./piper --assetPathList ./piper_master
+            --assetPathList ./piper-darwin.x86_64 --assetPathList ./piper_master-darwin.x86_64
+            --assetPathList ./piper-darwin.arm64 --assetPathList ./piper_master-darwin.arm64
+
+      - name: Download Piper binary from recently published prerelease
+        uses: robinraju/release-downloader@v1
+        with:
+          tag: ${{ env.PIPER_version }}
+          fileName: 'piper'
+
+      - name: Test binary (check output for 'commit:' substring)
+        run: |
+          chmod +x piper
+          if ./piper version | grep -Fq "commit:"; then
+              echo "piper binary test is successful"
+          else
+              echo "piper binary test failed"
+              ./piper version
```

Comment:
Why is here an additional `piper` call?

Reply:
to see what it prints instead of expected output